src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -442,9 +446,55 @@ bool leader_balancer::should_stop_balance() const {
 }
 
 bool leader_balancer::leadership_pinning_enabled() const {
+    if (_feature_table.should_sanction()) {
+        const auto warning_msg = leadership_pinning_sanction_warning_msg();
+        if (warning_msg.has_value()) {
+            vlog(clusterlog.warn, "{}", warning_msg.value());
+        }
+        return false;
+    }
+
     return _enable_rack_awareness();
 }
 
+std::optional<ss::sstring>
+leader_balancer::leadership_pinning_sanction_warning_msg() const {
+    const auto& [_, config_prop_sanctioned] = _default_preference(true);
+    chunked_vector<model::topic_namespace> sanctioned_topics;
+
+    for (const auto& topic : _topics.topics_map()) {
+        const auto& preference
+          = topic.second.get_configuration().properties.leaders_preference;
+        if (
+          preference.has_value()
+          && config::shard_local_cfg()
+               .default_leaders_preference.check_restricted(
+                 preference.value())) {
+            sanctioned_topics.push_back(topic.first);
+        }
+    }
+
+    if (!config_prop_sanctioned && sanctioned_topics.empty()) {
+        return std::nullopt;
+    }
+
+    ss::sstring msg
+      = "A Redpanda Enterprise Edition license is required to use enterprise "
+        "feature \"leadership pinning\". This feature is disabled.";
+    if (config_prop_sanctioned) {
+        msg += ssx::sformat(
+          " Value of property \"{}\" is being ignored.",
+          config::shard_local_cfg().default_leaders_preference.name());
+    }
+    if (!sanctioned_topics.empty()) {
+        msg += ssx::sformat(
+          " Value of topic property \"redpanda.leaders.preference\""
+          " in topics \"{}\" is being ignored.",
```

Comment:
nit: probably better to limit the number of printed topics to ~10 to avoid absurdly long log messages.

Reply:
If this is a situation where we could cause an overallocation warning I'd be fine with limiting it.  If it's just because "it's annoying", then I'd say leave it.  The sanctions are there to be annoying :)

Reply:
the way it is implemented, it could cause an overallocation, as `ss::sstring` seems to be contiguous. I'll have a look to see if this is easy to work around, otherwise i will limit it.